fea-rs/src/compile/lookups/gpos_builders.rs

```diff
@@ -153,19 +153,16 @@ impl ClassPairPosBuilder {
         class2: GlyphSet,
         record2: ValueRecord,
     ) {
-        let key = (record1.format(), record2.format());
-        let entry = self.0.entry(key).or_default();
-        let add_sub = match entry.last() {
+        let needs_new_subtable = match self.0.last() {
             None => true,
             Some(subtable) => !subtable.can_add(&class1, &class2),
         };
-        if add_sub {
-            entry.push(Default::default());
+        if needs_new_subtable {
+            self.0.push(Default::default());
+        }
+        if let Some(subtable) = self.0.last_mut() {
+            subtable.add(class1, class2, record1, record2);
         }
```

Comment:
At a glance this seems to have more branching than strictly necessary?

```rust
// If we don't have any subtables or we can't add to the last one make a new one
if !self.0.last().map(|s| s.can_add(...).unwrap_or_default() {
  self.0.push(...)
}
// We definitely have a last now
self.0.last_mut().add(...)

Reply:
I wrote it this way instead of using an unwrap, but we could definitely safely unwrap.

Reply:
Note that this replaces match/if/if not *only* the final unnecessary if